docs/RELEASE.md

```diff
@@ -0,0 +1,175 @@
+# Release process
+
+The release process for nobl9-go cannot be discussed without the context of
+our platform releases. In the following document, `n9` will refer to our
+platform **production** version.
+
+## The process
+
+1. Create release candidate version. \
+  Most likely this step is where you can stop.
+  Creating an official version release is only necessary if you want our users
+  to have immediate access to the released changes.
+  In many scenarios it might not be the case.
+  Furthermore, creating release candidate version instead of official release
+  gives us more time to "sit" on the introduced changes, which in turn
+  increases a chance for bug detection and gives room to introduce breaking
+  changes which might be necessary to address the uncovered issue.
+2. Create official release version (restricted access). \
+  Reach out to Foundations domain if you want to go with the official release.
+
+## Merging to main
+
+**NEVER** merge commits or tags based versions of n9 if they are based off of
+any other branch then main.
+Rephrasing, you can **ONLY** merge SDK reference to n9 main branch if it
+points at SDK main branch.
+
+The single, most important rule to keep in mind when merging to nobl9-go main
+branch is that it **MUST** be release ready. You **CANNOT** assume that
+a release is going to happen on X date. There might be bugs or features you're
+not aware of which will require immediate release.
+
+How should I then account for different scenarios in relation to n9 state?
+Below, we're listing the most common scenarios.
+If you find this list lacking in any manner, please reach out so that we can
+discuss your case and eventually write it down in this list.
+
+### The changes are not required to be in sync with n9
+
+In this scenario, there's nothing blocking you from merging to main.
+
+<!-- markdownlint-disable-next-line MD001 -->
+##### Examples
+
+- Bug fix which corrects SDK specific behavior.
+- New endpoint which is already available through n9 API.
+- Dependency update.
+- Documentation enhancement.
+
+### The changes are required to be in sync with n9 (backwards compatible)
+
+This scenario can be divided into two cases, depending on the following
+characteristics of the code you've written:
+
+- Visibility, is it exported (available through the public contract of the SDK)
```

Comment:
not sure if 'visibility' is a good word here, it's more about 'exposure' or 'accessibility' or just 'Public / Private', 
the most common wording imho is 'public / private access modifiers' but on the other hand accessibility is a term that is used for assisted technologies and things like WCAG standard

Reply:
isn't this just about changing the current public contract?